torch/csrc/autograd/python_saved_variable_hooks.cpp

```diff
@@ -44,4 +44,35 @@ namespace torch { namespace autograd {
       Py_XDECREF(data_);
     }
   }
+
+  // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
```

Comment:
Not sure if there's a better way to do this

Reply:
So the main issue with this is that if multiple threads try to use your context manager at the same time, there are going to be conflicts where they set the same thing.
Note that this is most likely ok if you raise an error if someone tries to set the hooks while they're already set.